client/galaxy/scripts/mvc/dataset/dataset-li-edit.js

```diff
@@ -60,12 +60,10 @@ var DatasetListItemEdit = _super.extend(
             deleted = this.model.get( 'deleted' ),
             editBtnData = {
                 title       : _l( 'Edit attributes' ),
-                href        : this.model.urls.edit,
-                target      : this.linkTarget,
+                href        : Galaxy.root + 'datasets/edit?dataset_id=' + this.model.attributes.dataset_id,
```

Comment:
If you want to build the url this way the correct id is `this.model.attributes.id`